cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringStore.java

```diff
@@ -174,6 +179,18 @@ public void updatePeerTimestamp(String peerName) throws IOException {
    * @throws PeerNotFoundException if the peer is not found
    */
   public void deletePeer(String peerName) throws IOException, PeerNotFoundException {
+    deletePeer(peerName, false);
+  }
+
+  /**
+   * Deletes a peer
+   *
+   * @param peerName name of the peer
+   * @param deleteSubscriberState whether to delete subscriber state
+   * @throws IOException if deleting the table fails
+   * @throws PeerNotFoundException if the peer is not found
+   */
+  public void deletePeer(String peerName, boolean deleteSubscriberState) throws IOException, PeerNotFoundException {
```

Comment:
when would you want to keep the subscriber state while deleting the peer?

Reply:
Subscriber state is not stored on the server side (client passes last message id to read from, so we don't maintain that state on the server). So we don't need to delete subscriber state on the server.